### PR TITLE
Whitespace in For statements

### LINKED ISSUE #7: {{For}} cycle does not preserve template layout
Presence or absence of the empty line(s) before the {{for}} cycle affects contents of the body of the cycle

Please see an example at https://gist.github.com/slavaschmidt/f0b93957f6f26dbab4e8

### LINKED ISSUE #24: Beard trims the spaces right after the interpolations in {{for}} statements
This Beard template:

``` beard
{{for user in users}}
<div>Hello {{user.name}}</div>
{{/for}}
```

is rendering:

``` html
<div>HelloDan</div>
<div>HelloCesar</div>
```

### PR DESCRIPTION
**Pull request by danpersa**
Closes #7 
Closes #24

### CODE REVIEW

**Comment by slavaschmidt** [comment]
@danpersa Looks good, what would you like me to do with the PR?

**Comment by danpersa** [comment]
Just review the code :) see if it looks good :)

**Comment by slavaschmidt** [comment]
Ok, done :+1:

File: src/test/resources/whitespaces/for-example.rendered [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <div>Gigi</div><div>Gicu</div>
+</div>
+
+<div class="second">
+  <div>Hello Gigi</div>
+  <div>Hello Gicu</div>
+</div>
+
+<div class="third">
+
```
Comment (rcillo) [inline src/test/resources/whitespaces/for-example.rendered:13]: Is this newline correct? I thought it should be

```
<div class="third">
  <div>{{user.name}}</div>  <div>{{user.name}}</div>
</div>
```

Comment (danpersa) [inline src/test/resources/whitespaces/for-example.rendered:13]: it's correct, because the closing of the `{{/for}}` is not on a new line.

**Pull request merged**